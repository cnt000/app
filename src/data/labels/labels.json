{
  "openMenuLabel": "apri il menu",
  "title": "Pungilandia",
  "titleDescription": "Piante grasse, strane ed insolite",
  "goToCartLabel": "vai al carrello",
  "copyright": "Copyright © 2020",
  "privacy": "privacy",
  "contact": "contatti",
  "company": "Azienda Agricola Pungilandia Rocchi Domenico",
  "companyAddress": "Via Isonzo 11, 65123 Pescara P.IVA 01203140684 CF RCCDNC59R30I712B\nIscrizione REA Pescara Nr.97422",
  "disclaimer": "Azienda in possesso Registro di Detenzione di Specie Vegetali Incluse\nNegli Allegati CITES A e B del Regolamento (CE) 338/97 del Consiglio del 9\nDicembre 1996 e Successive Modifiche e Attuazioni. Registro N.000001/2001\nVAB",
  "mailToLink": "mailto:info@pungilandia.com",
  "privacyLink": "http://www.pungilandia.com",
  "addToCart": "Aggiungi al carrello",
  "searchPlants": "Cerca piante",
  "searchPlantsImage": "http://www.pungilandia.com/php/images/lo/cotyl_mediop.jpg",
  "hintsFromPungilandiaTitle": "I Consigli di Pungilandia",
  "articleBy": "A cura del dott. Mimmo Rocchi",
  "articleContent": "consigliColtivazione.md",
  "mimmoImage": "http://www.pungilandia.com/media/MIMMO.JPG",
  "menuChiSiamo": "Chi siamo",
  "menuListaPiante": "Lista piante",
  "menuNovita": "Novità",
  "menuProdottiSpeciali": "Prodotti speciali",
  "menuConsigliPerColtivare": "Consigli per coltivare",
  "menuOrdini": "Ordini",
  "menuFotoShow": "Foto show",
  "menuEmail": "E-mail"
}
