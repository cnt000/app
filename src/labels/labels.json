{
  "openMenuLabel": "apri il menu",
  "title": "Pungilandia",
  "titleDescription": "Piante grasse, strane ed insolite",
  "goToCartLabel": "vai al carrello",
  "copyright": "Copyright Â© 2020",
  "privacy": "privacy",
  "contact": "contatti",
  "company": "Azienda Agricola Pungilandia Rocchi Domenico",
  "companyAddress": "Via Isonzo 11, 65123 Pescara P.IVA 01203140684 CF RCCDNC59R30I712B<br>Iscrizione REA Pescara Nr.97422",
  "disclaimer": "Azienda in possesso Registro di Detenzione di Specie Vegetali Incluse<br>Negli Allegati CITES A e B del Regolamento (CE) 338/97 del Consiglio del 9<br>Dicembre 1996 e Successive Modifiche e Attuazioni. Registro N.000001/2001<br>VAB"
}
