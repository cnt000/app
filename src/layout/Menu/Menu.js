import React, { useContext } from 'react';
import PropTypes from 'prop-types';

import LabelsContext from '../../features/labels';
import styles from './Menu.module.css';

const Menu = ({ isOpen, setMenuOpen, Link }) => {
  const {
    menuChiSiamo,
    menuListaPiante,
    menuNovita,
    menuProdottiSpeciali,
    menuConsigliPerColtivare,
    menuOrdini,
    menuFotoShow,
    menuEmail,
  } = useContext(LabelsContext);
  return (
    <>
      <nav className={`${styles.menu} ${isOpen ? styles.open : ''}`}>
        <div className={styles.header}>
          <button className={styles.close} onClick={setMenuOpen}>
            chiudi il menu
          </button>
        </div>
        <ul className={styles.entries}>
          <li>
            <Link to="/">{menuChiSiamo}</Link>
          </li>
          <li>
            <Link to="/plp">{menuListaPiante}</Link>
          </li>
          <li>
            <Link to="/pdp">{menuNovita}</Link>
          </li>
          <li>
            <Link to="/article">{menuProdottiSpeciali}</Link>
          </li>
          <li>
            <Link to="/test2">{menuConsigliPerColtivare}</Link>
          </li>
          <li>
            <Link to="/test3">{menuOrdini}</Link>
          </li>
          <li>
            <Link to="/test4">{menuFotoShow}</Link>
          </li>
          <li>
            <Link to="/test5">{menuEmail}</Link>
          </li>
        </ul>
      </nav>
      <div className={styles.overlay}></div>
    </>
  );
};

Menu.propTypes = {
  isOpen: PropTypes.bool,
  setMenuOpen: PropTypes.func,
};

export default Menu;
