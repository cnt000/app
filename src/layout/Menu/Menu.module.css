.menu {
  grid-area: sidebar;
  width: 14rem;
  max-width: 14rem;
  height: 100vh;
  position: absolute;
  top: 0;
  left: calc(-14rem - 2px);
  transition: left 0.4s;
  display: grid;
  grid-template-rows: 3.2rem 1fr;
  grid-template-areas:
    'header'
    'entries';
  background: var(--ochre);
  border-right: 2px dotted;
  border-image: linear-gradient(180deg, var(--green), var(--orange) 30em) 20;
  z-index: 2;
}

.open {
  left: 0;
  position: fixed;
}

.menu .entries li {
  padding: 0.6rem 0;
  text-align: center;
  font-family: 'Caveat Brush', sans-serif;
  font-size: 20px;
  text-shadow: 2px 2px 16px var(--grey);
}

.menu .entries a,
.menu .entries a:visited {
  text-decoration: none;
  letter-spacing: 1px;
  color: var(--green);
}

.menu .entries a:hover {
  color: var(--white);
}

.menu .close {
  appearance: button;
  background-color: transparent;
  text-decoration: none;
  color: initial;
  cursor: pointer;
  text-indent: 100%;
  white-space: nowrap;
  overflow: hidden;
  padding: 0;
  border: 0;
  background-image: url('../../images/close.svg');
  background-repeat: no-repeat;
  display: block;
  text-indent: -9999px;
  width: 3rem;
  height: 2.4rem;
  background-size: 3.1rem 1.9rem;
  margin-top: 0.7rem;
  margin-right: 0.4rem;
}

.overlay {
  display: none;
}

.menu.open + .overlay {
  display: block;
  position: fixed;
  top: 0;
  left: 0;
  width: 100vw;
  height: 100vh;
  background-color: transparent;
  z-index: 1;
}

@media only screen and (min-width: 1025px) {
  .menu {
    position: static;
    width: auto;
    height: auto;
    z-index: 1;
    padding-top: 2em;
  }

  .menu header {
    display: none;
  }

  .menu .entries {
    margin-top: 1rem;
  }
}
