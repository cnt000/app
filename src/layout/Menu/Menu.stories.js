import React from 'react';
import { action } from '@storybook/addon-actions';
import { BrowserRouter as Router, Link } from 'react-router-dom';

import Menu from './Menu';
import labels from '../../data/labels/labels.json';
import LabelsContext from '../../features/labels';

export default {
  component: Menu,
  title: 'Menu',
};

const actionsData = {
  setMenuOpen: action('setMenuOpen(false)'),
};

export const Default = () => (
  <LabelsContext.Provider value={labels}>
    <Router>
      <Menu isOpen={true} Link={Link} {...actionsData} />
    </Router>
  </LabelsContext.Provider>
);
export const Closed = () => (
  <LabelsContext.Provider value={labels}>
    <Router>
      <Menu isOpen={false} Link={Link} {...actionsData} />
    </Router>
  </LabelsContext.Provider>
);
