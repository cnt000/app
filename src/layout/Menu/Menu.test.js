import React from 'react';
import ReactDOM from 'react-dom';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { act } from 'react-dom/test-utils';
import { BrowserRouter as Router } from 'react-router-dom';

import Menu from './Menu';
import labels from '../../data/labels/labels.json';
import LabelsContext from '../../features/Labels';

const setMenuOpenMock = jest.fn();

const menuComponentWithLabels = (
  <LabelsContext.Provider value={labels}>
    <Menu isOpen={true} setMenuOpen={setMenuOpenMock} />
  </LabelsContext.Provider>
);

describe('Menu', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('renders correctly', () => {
    const tree = renderer
      .create(<Router>{menuComponentWithLabels}</Router>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Click ', () => {
  let container;

  beforeEach(() => {
    container = document.createElement('div');
    document.body.appendChild(container);
  });

  afterEach(() => {
    document.body.removeChild(container);
    container = null;
  });

  it('click close button call setMenuOpenMock', () => {
    const setMenuOpenMockWE = jest.fn();

    act(() => {
      ReactDOM.render(
        <Router>
          <Menu isOpen={true} setMenuOpen={setMenuOpenMockWE} />
        </Router>,
        container,
      );
    });
    const button = container.querySelector('button.close');

    act(() => {
      button.dispatchEvent(new MouseEvent('click', { bubbles: true }));
    });
    expect(setMenuOpenMockWE).toHaveBeenCalled();
  });

  it('has class open if isOpen is true', () => {
    const setMenuOpenMockWE = jest.fn();

    act(() => {
      ReactDOM.render(
        <Router>
          <Menu isOpen={true} setMenuOpen={setMenuOpenMockWE} />
        </Router>,
        container,
      );
    });
    expect(container.querySelectorAll('.open').length).toBe(1);
  });

  test('renders home', () => {
    const { getByText } = render(<Router>{menuComponentWithLabels}</Router>);
    expect(getByText(/Home/)).toBeInTheDocument();
  });
});
