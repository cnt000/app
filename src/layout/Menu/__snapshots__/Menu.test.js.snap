// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu renders correctly 1`] = `
Array [
  <nav
    className="menu open"
  >
    <div
      className="header"
    >
      <button
        className="close"
        onClick={[MockFunction]}
      >
        chiudi il menu
      </button>
    </div>
    <ul
      className="entries"
    >
      <li>
        <a
          href="/"
          onClick={[Function]}
        >
          Chi siamo
        </a>
      </li>
      <li>
        <a
          href="/plp"
          onClick={[Function]}
        >
          Lista piante
        </a>
      </li>
      <li>
        <a
          href="/pdp"
          onClick={[Function]}
        >
          Novit√†
        </a>
      </li>
      <li>
        <a
          href="/article"
          onClick={[Function]}
        >
          Prodotti speciali
        </a>
      </li>
      <li>
        <a
          href="/test2"
          onClick={[Function]}
        >
          Consigli per coltivare
        </a>
      </li>
      <li>
        <a
          href="/test3"
          onClick={[Function]}
        >
          Ordini
        </a>
      </li>
      <li>
        <a
          href="/test4"
          onClick={[Function]}
        >
          Foto show
        </a>
      </li>
      <li>
        <a
          href="/test5"
          onClick={[Function]}
        >
          E-mail
        </a>
      </li>
    </ul>
  </nav>,
  <div
    className="overlay"
  />,
]
`;
